src/XR.tsx

```diff
@@ -112,44 +123,34 @@ export function XR(props: { children: React.ReactNode }) {
     }
   }, [gl])
 
+  React.useEffect(() => {
+    if (buttonVR) {
+      const child = document.body.appendChild(VRButton.createButton(gl))
+      return () => { document.body.removeChild(child); }
+    }
+  }, [gl, buttonVR]);
+
+  React.useEffect(() => {
+    if (buttonAR) {
+      const child = document.body.appendChild(ARButton.createButton(gl, sessionInit))
+      return () => { document.body.removeChild(child); }
+    }
+  }, [gl, buttonAR, sessionInit])
+
   const value = React.useMemo(() => ({ controllers, isPresenting, player }), [controllers, isPresenting, player])
 
   return (
     <XRContext.Provider value={value}>
       <primitive object={player} dispose={null}>
         <primitive object={camera} dispose={null} />
       </primitive>
-      {props.children}
+      <InteractionManager>
+        {children}
+      </InteractionManager>
     </XRContext.Provider>
   )
 }
 
-function XRCanvas({ children, ...rest }: ContainerProps) {
-  return (
-    <Canvas vr {...rest}>
-      <XR>
-        <InteractionManager>{children}</InteractionManager>
-      </XR>
-    </Canvas>
-  )
-}
-
-export function VRCanvas({ children, ...rest }: ContainerProps) {
```

Comment:
I think VR and AR canvas components can stay with all the `Canvas frameloop="never" onCreated={({ gl }) => startXRloop(gl)}` and AR/VR buttons predefined, not only for compatibility but also for convenience. It'd be unfortunate to have users make their code more verbose